opencog/server/BaseServer.h

```diff
@@ -65,8 +65,41 @@ class BaseServer
     virtual ~BaseServer(void);
 }; // class
 
-// singleton instance (following meyer's design pattern)
+// Singleton instance (following meyer's design pattern) NOTE: There are
+// problems with the way this is being used, in some cases, as it is confusing
+// that there are examples of both stack based and factory function singleton
+// server creation in the tests and examples. For example, I saw some code
+// that did this:
+//
+// 	  void testAddPunishmentPredicate() {
+//        server(SpaceTimeCogServer::createInstance);  << creates one instance
+//        SpaceTimeCogServer srv;                      << creates another
+//        AtomSpace& atomSpace = srv.getAtomSpace();
+//
```

Comment:
Yes, all this needs fixing.  Its not at all clear what the correct fix should be.  The space server may need to be replaced by or integrated with a ROS-based space server, so that spatial and motion-planning information can be shared with robot platforms. This is still wildly unclear to me.

The Embodiment server inherits from the CogServer, but I don't know of any reason that it should. I don't see how it derives any benefit from doing so. It seems like some old arbitrary design decision.  Again, its not clear to me what the correct fix should be.